File: Pokemanos.Model/Usuario.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Pokemanos.Model
{
    public class Usuario
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "O campo {0} é Obrigatório")]
        [MinLength(2)]
        [MaxLength(100)]
        public string Nome { get; set; }

        [Required(ErrorMessage = "O campo {0} é Obrigatório")]
        [MinLength(2)]
        [MaxLength(100)]
        public string Sobrenome { get; set; }

        [Required(ErrorMessage = "O campo {0} é Obrigatório")]
        [EmailAddress]
        [Display(Name = "E-mail", Prompt = "Digite o {0}")]
        public string Email { get; set; }
      
        [Required(ErrorMessage = "O campo {0} é Obrigatório")]
        [Display(Name = "Telefone", Prompt = "Digite o {0}")]
        public string Senha { get; set; }

        [Required(ErrorMessage = "O campo {0} é Obrigatório")]
        [Phone]
        [Display(Name = "Telefone", Prompt = "Digite o {0}")]
        public string Telefone { get; set; }

        [Required(ErrorMessage = "O campo {0} é Obrigatório")]
        [DataType(DataType.DateTime)]
        [Display(Name = "Data", Prompt = "Digite o {0}")]
        public DateTime DataCadastro { get; set; }

        public CodigoSeguranca CodigoSeguranca { get; set; }
    }
}

File: Pokemanos.Web/src/app/core/model/usuario.ts
export class Usuario {
  constructor() {
    this.id = 0;
    this.nome = '';
    this.sobrenome = '';
    this.email = '';
    this.senha = '';
    this.telefone = '';
    this.dataCadastro = new Date();
  }

  id: number;
  nome: string;
  sobrenome: string;
  email: string;
  senha: string;
  telefone: string;
  dataCadastro: Date;
}

export class Login {
  constructor() {
    this.email = '';
    this.senha = '';
    this.lembrar = false;
  }

  email: string;
  senha: string;
  lembrar: boolean;
}
